spec/hedgelog_spec.rb

```diff
@@ -1,11 +1,300 @@
 require 'spec_helper'
+require 'timecop'
+require 'benchmark'
+require 'json'
 
 describe Hedgelog do
   it 'has a version number' do
     expect(Hedgelog::VERSION).not_to be nil
   end
 
-  it 'does something useful' do
-    expect(false).to eq(true)
+  let(:log_dev) { StringIO.new }
+  let(:log_level) { Logger::INFO }
+  let(:logger) { Hedgelog.new(log_dev) }
+  let(:log_exec) { -> { logger.debug 'foo' } }
+  let(:log_results) { log_dev.string }
+  subject do
+    log_exec.call
+    logger.level = log_level
+    JSON.parse(log_results)
+  end
+
+  %w(debug info warn error fatal unknown).each do |level|
+    describe "\##{level}" do
+      let(:log_level) { level.to_sym }
+      before :each do
+        Timecop.freeze(2015, 01, 01)
+      end
+
+      after :each do
+        Timecop.return
+      end
+
+      context 'when log input is the string "FOO" the output' do
+        let(:log_exec) { -> { logger.fatal 'FOO' } }
+
+        it do
+          should include(
+            'message' => 'FOO',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.